middlewares/accesslog/save_backend.go

```diff
@@ -0,0 +1,71 @@
+package accesslog
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// SaveBackend sends the backend name to the logger. These are always used with a corresponding
+// SaveFrontend handler.
+type SaveBackend struct {
+	next        http.Handler
+	backendName string
+}
+
+// NewSaveBackend creates a SaveBackend handler.
+func NewSaveBackend(next http.Handler, backendName string) http.Handler {
+	return &SaveBackend{next, backendName}
+}
+
+func copyHeaders(src http.Header) http.Header {
```

Comment:
Our vulcand/oxy dependency already [brings a header copying functionality](https://github.com/containous/traefik/blob/d8c21639f79573f5e0e48bf509bf296f7916c0d9/vendor/github.com/vulcand/oxy/utils/netutils.go#L109). Not sure if you can use it though since it does not override headers.